cmd/dep/root_analyzer.go

```diff
@@ -139,6 +140,26 @@ func (a *rootAnalyzer) FinalizeRootManifestAndLock(m *dep.Manifest, l *dep.Lock)
 			delete(m.Constraints, pr)
 		}
 	}
+	// Pick the direct dependencies from the solution lock and add to manifest.
+	// This is done to fill up the manifest constraints with the dependencies
+	// solved over the network.
+	for _, y := range l.Projects() {
+		var f *fb.ConstraintFeedback
+		pr := y.Ident().ProjectRoot
+		if _, ok := a.directDeps[string(pr)]; ok {
+			pp := getProjectPropertiesFromVersion(y.Version())
+			if pp.Constraint != nil {
+				m.Constraints[pr] = pp
+				pc := gps.ProjectConstraint{Ident: y.Ident(), Constraint: pp.Constraint}
+				f = fb.NewConstraintFeedback(pc, fb.DepTypeDirect)
+			} else {
+				f = fb.NewLockedProjectFeedback(y, fb.DepTypeDirect)
+			}
+		} else {
+			f = fb.NewLockedProjectFeedback(y, fb.DepTypeTransitive)
+		}
+		f.LogFeedback(a.ctx.Err)
+	}
```

Comment:
Problem with this is that, since `rootAnalyzer` isn't aware of network/gopath mode of execution, even in gopath mode, these feedbacks would run. Resulting in double feedbacks.

Should this be moved to init? or should we make rootAnalyzer aware of gopath flag?